providers/elbv1/elbv1svc/elb_classic.go

```diff
@@ -0,0 +1,362 @@
+			aws.String(loadBalancerName),
+		},
+		Tags: []*elb.TagKeyOnly{
+			{
+				Key: aws.String(tagKey),
+			},
+		},
+	}
+	_, err := svc.elbc.RemoveTags(params)
+	if err != nil {
+		return fmt.Errorf("RemoveTags SDK error: %v", err)
+	}
+
+	return nil
+}
+
+// EnsureListenerInstancePort ensures that the specified listener
+// is configured to forward traffic to the specified instance port.
+// If not, it recreates the listener and configures it so, while
+// maintaining all it's other properties.
+func (svc *ELBClassicService) EnsureListenerInstancePort(
+	loadBalancerName string, instancePort int64, listenerDesc *elb.ListenerDescription) error {
+	logrus.Debugf("EnsureListenerInstancePort => name: %s, port: %d, listener: %v",
+		loadBalancerName, instancePort, listenerDesc)
+
+	loadBalancerPort := *listenerDesc.Listener.LoadBalancerPort
+	if *listenerDesc.Listener.InstancePort == instancePort {
+		logrus.Debugf("Listener %s:%d is already using instance port %d",
+			loadBalancerName, loadBalancerPort, instancePort)
+		return nil
+	}
+
+	logrus.Debugf("Recreating listener %s:%d with instance port %d",
+		loadBalancerName, loadBalancerPort, instancePort)
+
+	{
+		params := &elb.DeleteLoadBalancerListenersInput{
+			LoadBalancerName: aws.String(loadBalancerName),
+			LoadBalancerPorts: []*int64{
+				aws.Int64(loadBalancerPort),
+			},
+		}
+
+		_, err := svc.elbc.DeleteLoadBalancerListeners(params)
+		if err != nil {
+			return fmt.Errorf("DeleteLoadBalancerListeners SDK error: %v", err)
+		}
+	}
+
+	{
+		params := &elb.CreateLoadBalancerListenersInput{
+			Listeners: []*elb.Listener{
+				{
+					InstancePort:     aws.Int64(instancePort),
+					LoadBalancerPort: aws.Int64(loadBalancerPort),
+					Protocol:         listenerDesc.Listener.Protocol,
+					InstanceProtocol: listenerDesc.Listener.InstanceProtocol,
+					SSLCertificateId: listenerDesc.Listener.SSLCertificateId,
+				},
+			},
+			LoadBalancerName: aws.String(loadBalancerName),
+		}
+
+		_, err := svc.elbc.CreateLoadBalancerListeners(params)
+		if err != nil {
+			return fmt.Errorf("CreateLoadBalancerListeners SDK error: %v", err)
+		}
+	}
+
+	logrus.Debugf("Restoring policies for Listener %s/%d: %v",
+		loadBalancerName, loadBalancerPort, listenerDesc.PolicyNames)
+
+	// Restoring policies
+	if len(listenerDesc.PolicyNames) > 0 {
+		params := &elb.SetLoadBalancerPoliciesOfListenerInput{
+			LoadBalancerName: aws.String(loadBalancerName),
+			LoadBalancerPort: aws.Int64(loadBalancerPort),
+			PolicyNames:      listenerDesc.PolicyNames,
+		}
+
+		_, err := svc.elbc.SetLoadBalancerPoliciesOfListener(params)
+		if err != nil {
+			return fmt.Errorf("SetLoadBalancerPoliciesOfListener SDK error: %v", err)
+		}
+	}
+
+	return nil
+}
+
+// EnsureHealthCheckPort ensures that the specified healthcheck
+// uses the specified instance port. If not, it configures it
+// so, while maintaining all it's other properties.
+func (svc *ELBClassicService) EnsureHealthCheckPort(
```

Comment:
Is this called from the external_lb provider(aws_elbv1.go)? Could not find any references